src/loader.ts

```diff
@@ -178,6 +178,12 @@ export async function loadConfig<
     }
   }
 
+  // !IMPORTANT: Needs to be after resolving config to then apply merge strategy
+  // Use merge strategy if provided by user
+  if (options.mergeStrategy && Object.keys(options.mergeStrategy).length > 0) {
+    r.config = defuFn(options.mergeStrategy, r.config) as T;
```

Comment:
I _guess_ we want to do it for merging layers (in `defu` above) because currently while it is a good feature we are practically implementing **per key normalization** result is already merged before.